src/share_policy.rs

```diff
@@ -0,0 +1,37 @@
+use std::fmt::Debug;
+
+use futures::Future;
+
+use super::{DocHandle, PeerId};
+
+pub enum ShareDecision {
+    Share,
+    DontShare,
+}
+
+/// A policy for determining whether to share a document with a peer
+pub trait SharePolicy: Clone + Debug {
+    type Error: std::error::Error + Send + 'static;
+    type DecisionFuture: Future<Output = Result<ShareDecision, Self::Error>>;
+
+    /// Determine whether to share a document with a peer
+    fn should_share(&self, doc: DocHandle, peer: PeerId) -> Self::DecisionFuture;
+}
```

Comment:
I think that's an interesting API, and something that can be explored later, because the default "sync with all known peers" seems like a good first step to get right(and a share policy of sorts could even be achieved with that default by using multiple repos).